create data semi-structured

Matheus1714 opened the pull request.

Matheus1714 commented:
Images to compare

![Captura de tela 2023-08-02 194849](https://github.com/ArturAssisComp/pizzaiolo/assets/39354089/5c917190-d4f8-4923-84d9-e0251551af98)
![Captura de tela 2023-08-02 194912](https://github.com/ArturAssisComp/pizzaiolo/assets/39354089/e6150d4e-7f8c-4b3e-8e15-a50ed2c99d6e)

ArturAssisComp commented on data/dadosPizzas.json:
```diff
@@ -0,0 +1,508 @@
+{
+    "pizzas":[
+        {
+            "_id":1,
```
Create another field (Suggestion: "numero_cardapio") to store the number of the pizza mapped to the menu. Use the field "_id" to store the hash generated id.

ArturAssisComp commented on data/dadosPizzas.json:
```diff
@@ -0,0 +1,508 @@
+            "nome": "Três Garotos",
+            "preco":{
+                "grande": 48.00,
+                "gidante": 96.00
```
Fix the typo: Instead of "gidante", use "gigante".

ArturAssisComp commented on data/dadosPizzas.json:
```diff
@@ -0,0 +1,508 @@
+                "grande": 48.00,
+                "gidante": 96.00
+            },
+            "acompanhamentos":"Presunto, Milho, Cebola, Ovos, Tomates, Mussarela, Calabresa e Bacon"
```
Instead of storing the ingredients as a unique string, create a list of strings. This will make it possible for us to handle the ingredients in an easier way (for example, if the user wants to exclude the ingredient Milho, it will be easier to specify that in the order message).

ArturAssisComp commented on data/dadosPizzas.json:
```diff
@@ -0,0 +1,508 @@
+            },
+            "acompanhamentos":"Mussarela, Tomate e Bacon"
+        },
+        {
```
The pizza "Sete Estações" is a different type of pizza. It is interesting to add a field "tipo" with the following options: "normal", "escolha-ingredientes", "doce". The structure of a "normal" pizza is 
```JSON
{
    "_id":"some_id",
    "tipo":"normal",
    "numero_cardapio":1,
    "nome":"some_name",
    "preco":{"grande":123, "gigante":234},
    "acompanhamentos":["ing1", "ing2"]
}
```
The structure of a "escolha-ingredientes" pizza is 
```JSON
{
    "_id":"some_id",
    "tipo":"escolha-ingredientes",
    "numero_cardapio":1,
    "nome":"some_name",
    "preco":{"grande":123, "gigante":234},
    "numero_de_ingredientes":7,
    "ingredientes_possiveis":["ing1", "ing2"]
}
```
The structure of a "doce" pizza is 
```JSON
{
    "_id":"some_id",
    "tipo":"doce",
    "numero_cardapio":1,
    "nome":"some_name",
    "preco":{"grande":123, "gigante":234},
    "acompanhamentos":["ing1", "ing2"]
}
```

Matheus1714 commented:
This is my code to format: 

```{python}
import json
import uuid
from typing import List

start_file = 'dadosPizzas.json'
final_file = 'dadosPizzas1.json'
key_pizzas = 'pizzas'

from ast import Return
def transform_to_list_of_dicts(input_string:str)->List[dict]:
  items = input_string.split(", ")
  result_list = []

  for item in items:
      sub_items = item.split(" e ")
      for sub_item in sub_items:
          result_list.append(sub_item.strip())

  return result_list

def get_tipo(datum:str)->dict:
  if datum['nome'] == 'Sete Estações':
    return 'escolha-ingredientes'
  elif datum['numero_cardapio'] >=54:
    return 'doce'
  else:
    return 'normal'

def format_datum(datum:dict)->dict:
  datum['numero_cardapio'] = datum['_id']
  datum['_id'] = uuid.uuid4().hex
  datum['acompanhamentos'] = transform_to_list_of_dicts(datum['acompanhamentos'])
  datum['tipo'] = get_tipo(datum)
  return datum

dados = []
with open(start_file, 'rb') as f:
  bin = f.read()
  text = bin.decode('utf-8')
  data = json.loads(text)

pizzas = data[key_pizzas]

new_pizzas = []
for pizza in pizzas:
  pizza = format_datum(pizza)
  new_pizzas.append(pizza)

with open(final_file, 'wb') as f:
  new_data = {key_pizzas: new_pizzas}
  text = json.dumps(new_data, sort_keys=True, indent=4)
  bin = text.encode('utf-8')
  f.write(bin)
```

Matheus1714 commented:
My code to to format the json was:
https://gist.github.com/Matheus1714/b147a6d71d2f2884315fcd3b0dc6eea4

ArturAssisComp commented on data/dadosPizzas.json:
```diff
@@ -0,0 +1,886 @@
+            "tipo": "normal"
+        },
+        {
+            "_id": "8b79271c9e624ab78385a0358a20a290",
```
Change the "sete-estações"
```JSON
{
    "_id":"some_id",
    "tipo":"escolha-ingredientes",
    "numero_cardapio":1,
    "nome":"some_name",
    "preco":{"grande":123, "gigante":234},
    "numero_de_ingredientes":7,
    "ingredientes_possiveis":["ing1", "ing2"]
}
```

ArturAssisComp reviewed (changes requested):
Change the 7stações
